packages/smart-contracts/src/contracts/ERC20SwapToPay.sol

```diff
@@ -82,15 +107,22 @@ contract ERC20SwapToPay is Ownable {
     IERC20 spentToken = IERC20(_path[0]);
     IERC20 requestedToken = IERC20(_path[_path.length-1]);
 
+    
     uint256 requestedTotalAmount = _amount + _feeAmount;
 
-    require(spentToken.allowance(msg.sender, address(this)) > _amountInMax,
+    // make sure the allowance includes the swapFee.
+    require(spentToken.allowance(msg.sender, address(this)) > (_amountInMax + _calculateSwapFee(requestedTotalAmount)),
```

Comment:
Because you imported SafeMath above, shouldn't it be .add instead of +
But I [think 0.8.0](https://blog.soliditylang.org/2020/12/16/solidity-v0.8.0-release-announcement/) also has the overflow checks ? so SafeMath is not required ?

Reply:
Yes, I used SafeMath.sol, it was a good habit I had, that turned into a bad habit after `0.8.0`, when `overflow` protection became standard from 0.8.0.
Do you want me to remove it? 
I haven't checked which requires most gas of `.add()` or `+`.
I think using SafeMath might be more expensive, so I´ll remove it. 
:-)

Reply:
Also, I didn't change any of the current code. so that´s why the +, etc is there. 

I wanted to change all `require(condition, string);` into `if () revert customError();` which would save some gas on error messages.

Reply:
I was just curious because I saw the import  but it wasn't used.
But about changing things and saving gas etc.., I'll leave that decision to Yoann or Benji, don't mind me 🙂